utils/joiner/tuplejoiner.cpp

```diff
@@ -166,7 +166,7 @@ TupleJoiner::TupleJoiner(const rowgroup::RowGroup& smallInput, const rowgroup::R
 TupleJoiner::TupleJoiner(const rowgroup::RowGroup& smallInput, const rowgroup::RowGroup& largeInput,
                          const vector<uint32_t>& smallJoinColumns, const vector<uint32_t>& largeJoinColumns,
                          JoinType jt, threadpool::ThreadPool* jsThreadPool, joblist::ResourceManager* rm,
-                         const uint64_t numCores)
+                         const uint64_t numCores, std::shared_ptr<std::array<std::optional<joblist::BlockedBloomFilter>, 2>> bloomFilters)
```

Comment:
I suggest you to alias this type b/c it is hard to read  shared_ptr<array<optional<something>>> TBH.